### PR TITLE
fix: [CDS-83638]: package dropdown for GAR

### PR DESCRIPTION
**Pull request by satyam-harness**
### Summary
Enhancement - Adding support of package field in GAR (https://harness.atlassian.net/browse/CDS-83638)
Solution - Added support of package in GAR in pipeline/runform/trigger.

Manual testing - 

Create form 

- Package is getting fetched when all required fields are fixed and have value.
- Package api call is not made when any of required field is runtime/expression or has empty fixed value.
- Clearing Package field when any of the required field is changed.
- Clearing Package dropdown list when any of the required field is changed.
- Package v1 api is called

Runtime form

- Package field is getting rendered when its made runtime in pipeline.
- Package is getting fetched when all required fields are fixed and have value.
- Package api call is not made when any of required field is runtime/expression or has empty fixed value.
- Clearing Package field when any of the required field is changed.
- Clearing Package dropdown list when any of the required field is changed.
- Package v2 api is called

Template form
- Changes made in Package field are reflected in yaml and unsaved changes doesnt appear once saved.
- Package is getting fetched when all required fields are fixed and have value.
- Package api call is not made when any of required field is runtime/expression or has empty fixed value.
- Clearing Package field when any of the required field is changed.
- Clearing Package dropdown list when any of the required field is changed.
- Package v1 api is called

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

https://github.com/harness/harness-core-ui/assets/96036463/22d74489-745a-4c42-a627-48cb911b6c29

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger sonar

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by satyam-harness** [comment]
trigger messagecheck

**Review by pankajsingh-harness — DISMISSED** [review]

**Review by Eric761 — DISMISSED** [review]

**Comment by bhavya-sinha** [comment]
LGTM

**Review by bhavya-sinha — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
looks good to me

File: cypress/fixtures/pipeline/api/triggers/Cypress_Test_Trigger_GAR_Package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        "garPackageDTOList": [
+            {
+                "packageName": "test-package",
+                "createTime": "2022-08-17T08:56:53.139215Z",
+                "updateTime": "2022-08-17T09:09:46.141240Z"
+            }
+        ]
+    },
+    "metaData": null,
+    "correlationId": "8b3f8160-33f8-440e-b8b0-9bbffb01faa1"
```
Comment (ishantmehtaharness) [inline cypress/fixtures/pipeline/api/triggers/Cypress_Test_Trigger_GAR_Package.json:13]: we should avoid adding real identifiers, better to add dumy identifiers

Comment (satyam-harness) [reply cypress/fixtures/pipeline/api/triggers/Cypress_Test_Trigger_GAR_Package.json:13]: sure, thanks

**Comment by satyam-harness** [comment]
retrigger jest

**Pull request merged**